### PR TITLE
Create azure storage for tf storage and key vault

### PR DESCRIPTION
**Pull request by sylvester-loreto-codurance**
Here create the azure storage and the key vault

### CODE REVIEW

**Comment by tasssko** [comment]
LGTM

File: workspace/rancher-bootstrap/main.tf [hunk 1]
```diff
@@ -1,8 +1,18 @@
 provider "azurerm" {
   version = "=2.30.0"
+  tenant_id = "7049e6a3-141d-463a-836b-1ba40d3ff653"
```
Comment (gargath) [inline workspace/rancher-bootstrap/main.tf:3]: This also needs the `subscrition_id`

File: .gitignore [hunk 2]
```diff
@@ -3,4 +3,136 @@
 *.tfstate.backup
 .terraform/*
 *.info
-*.tfvars
```
Comment (gargath) [inline .gitignore:6]: I personally do not like including `.tfvars` in git.
If you choose to do so, you MUST ensure at all times that no secrets are specified in input variables.

File: modules/azure/storage/variables.tf [hunk 3]
```diff
@@ -0,0 +1,29 @@
+variable "resource_group_name" {
```
Comment (gargath) [inline modules/azure/storage/variables.tf:1]: Should any of these have defaults maybe?

**Review by gargath — CHANGES_REQUESTED** [review]
Couple of minor nits.
One provider config is missing the Azure `subscription_id`

**Review by tasssko — APPROVED** [review]
Some valid concerns raised by @gargath however we can move with this now.

File: workspace/rancher-bootstrap/main.tf (hunk 1 above)
Comment (sylvester-loreto-codurance) [reply workspace/rancher-bootstrap/main.tf:3]: Done!

**Pull request merged**